src/GraphQlClientGenerator/GraphQlGenerator.cs

```diff
@@ -1023,8 +1096,8 @@ private void GenerateQueryBuilder(GenerationContext context, GraphQlType type, I
                 var field = fields[i];
                 var fieldType = field.Type.UnwrapIfNonNull();
                 var isList = fieldType.Kind == GraphQlTypeKind.List;
-                var treatUnknownObjectAsComplex = IsUnknownObjectScalar(type, field.Name, fieldType) && !_configuration.TreatUnknownObjectAsScalar;
-                var isComplex = isList || treatUnknownObjectAsComplex || IsComplexType(fieldType.Kind);
+                var treatUnknownObjectAsComplex = IsUnknownObjectScalar(type, field.Name, fieldType, field.AppliedDirectives) && !_configuration.TreatUnknownObjectAsScalar;
+                var isComplex = isList || treatUnknownObjectAsComplex || IsComplexType(fieldType.Kind) || field.Args?.Any() == true;

```

Comment:
I'm not sure this is the correct thing to do. Maybe we should call `WithAllScalarFields` + `Except***()` for scalars that have arguments